arrow-array/src/array/run_array.rs

```diff
@@ -0,0 +1,499 @@
+
+        let values_data = array.values();
+        assert_eq!(1, values_data.null_count());
+    }
+
+    #[test]
+    fn test_run_array_try_new() {
+        let values: StringArray = [Some("foo"), Some("bar"), None, Some("baz")]
+            .into_iter()
+            .collect();
+        let run_ends: Int32Array =
+            [Some(1), Some(2), Some(3), Some(4)].into_iter().collect();
+
+        let array = RunArray::<Int32Type>::try_new(&run_ends, &values).unwrap();
+        assert_eq!(array.run_ends().data_type(), &DataType::Int32);
+        assert_eq!(array.values().data_type(), &DataType::Utf8);
+
+        assert_eq!(array.run_ends.null_count(), 0);
+        assert_eq!(array.values().null_count(), 1);
+
+        assert_eq!(
+            "RunArray {run_ends: PrimitiveArray<Int32>\n[\n  1,\n  2,\n  3,\n  4,\n], values: StringArray\n[\n  \"foo\",\n  \"bar\",\n  null,\n  \"baz\",\n]}\n",
+            format!("{:?}", array)
+        );
+    }
+
+    #[test]
+    fn test_run_array_int16_type_definition() {
+        let array: Int16RunArray = vec!["a", "a", "b", "c", "c"].into_iter().collect();
+        let values: Arc<dyn Array> = Arc::new(StringArray::from(vec!["a", "b", "c"]));
+        assert_eq!(array.run_ends(), &Int16Array::from(vec![2, 3, 5]));
+        assert_eq!(array.values(), &values);
+    }
+
+    #[test]
+    fn test_run_array_empty_string() {
+        let array: Int16RunArray = vec!["a", "a", "", "", "c"].into_iter().collect();
+        let values: Arc<dyn Array> = Arc::new(StringArray::from(vec!["a", "", "c"]));
+        assert_eq!(array.run_ends(), &Int16Array::from(vec![2, 4, 5]));
+        assert_eq!(array.values(), &values);
+    }
+
+    #[test]
+    fn test_run_array_length_mismatch() {
+        let values: StringArray = [Some("foo"), Some("bar"), None, Some("baz")]
+            .into_iter()
+            .collect();
+        let run_ends: Int32Array = [Some(1), Some(2), Some(3)].into_iter().collect();
+
+        let actual = RunArray::<Int32Type>::try_new(&run_ends, &values);
+        let expected = ArrowError::InvalidArgumentError("The run_ends array length should be the same as values array length. Run_ends array length is 3, values array length is 4".to_string());
+        assert_eq!(
+            format!("{}", expected),
+            format!("{}", actual.err().unwrap())
+        );
+    }
+
+    #[test]
+    fn test_run_array_run_ends_with_null() {
+        let values: StringArray = [Some("foo"), Some("bar"), Some("baz")]
+            .into_iter()
+            .collect();
+        let run_ends: Int32Array = [Some(1), None, Some(3)].into_iter().collect();
+
+        let actual = RunArray::<Int32Type>::try_new(&run_ends, &values);
+        let expected = ArrowError::InvalidArgumentError("Found null values in run_ends array. The run_ends array should not have null values.".to_string());
+        assert_eq!(
+            format!("{}", expected),
+            format!("{}", actual.err().unwrap())
```

Comment:
```suggestion
            expected.to_string(),
            actual.unwrap_err().to_string()
```
And the same for below